packages/components/nodes/chains/GraphCypherQAChain/GraphCypherQAChain.ts

```diff
@@ -0,0 +1,223 @@
+            }
+        ]
+    }
+
+    async init(nodeData: INodeData): Promise<any> {
+        const model = nodeData.inputs?.model
+        const cypherModel = nodeData.inputs?.cypherModel
+        const qaModel = nodeData.inputs?.qaModel
+        const graph = nodeData.inputs?.graph
+        const cypherPrompt = nodeData.inputs?.cypherPrompt as BasePromptTemplate | FewShotPromptTemplate | undefined
+        const qaPrompt = nodeData.inputs?.qaPrompt as BasePromptTemplate | undefined
+        const returnDirect = nodeData.inputs?.returnDirect as boolean
+
+        // Handle prompt values if they exist
+        let cypherPromptTemplate: PromptTemplate | FewShotPromptTemplate | undefined
+        let qaPromptTemplate: PromptTemplate | undefined
+
+        if (cypherPrompt) {
+            if (cypherPrompt instanceof PromptTemplate) {
+                cypherPromptTemplate = new PromptTemplate({
+                    template: cypherPrompt.template as string,
+                    inputVariables: cypherPrompt.inputVariables
+                })
+                if (!qaPrompt) {
+                    throw new Error('QA Prompt is required when Cypher Prompt is a Prompt Template')
+                }
+            } else if (cypherPrompt instanceof FewShotPromptTemplate) {
+                const examplePrompt = cypherPrompt.examplePrompt as PromptTemplate
+                cypherPromptTemplate = new FewShotPromptTemplate({
+                    examples: cypherPrompt.examples,
+                    examplePrompt: examplePrompt,
+                    inputVariables: cypherPrompt.inputVariables,
+                    prefix: cypherPrompt.prefix,
+                    suffix: cypherPrompt.suffix,
+                    exampleSeparator: cypherPrompt.exampleSeparator,
+                    templateFormat: cypherPrompt.templateFormat
+                })
+            } else {
+                cypherPromptTemplate = cypherPrompt as PromptTemplate
+            }
+        }
+
+        if (qaPrompt instanceof PromptTemplate) {
+            qaPromptTemplate = new PromptTemplate({
+                template: qaPrompt.template as string,
+                inputVariables: qaPrompt.inputVariables
+            })
+        }
+
+        if ((!cypherModel || !qaModel) && !model) {
+            throw new Error('Model or is required when Cypher Model or QA Model are not provided')
+        }
+
+        // Validate required variables in prompts
+        if (!cypherPromptTemplate?.inputVariables.includes('schema') || !cypherPromptTemplate?.inputVariables.includes('question')) {
+            throw new Error('Cypher Generation Prompt must include {schema} and {question} variables')
+        }
+
+        const chain = GraphCypherQAChain.fromLLM({
+            llm: model,
+            qaLLM: qaModel,
+            cypherLLM: cypherModel,
+            graph,
+            cypherPrompt: cypherPromptTemplate,
```

Comment:
error when nothing is connected: 
![image](https://github.com/user-attachments/assets/d905224f-f5da-42a4-b19d-c062d512abed)